### PR TITLE
Add e2e support for vsphere

### PR DESCRIPTION
**Pull request by lleshchi**
This is the hive side of [HIVE-2270](https://issues.redhat.com/browse/HIVE-2270), which must be merged prior to its corresponding PR in the release repo. Adds support for vsphere platform in e2e tests, utilizing necessary environment variables gathered from the ci-operator workflow.

### CODE REVIEW

File: hack/e2e-common.sh
Comment (2uasimojo) [inline hack/e2e-common.sh:234]: Relies on two variables that we're not defaulting or validating. I don't love `set -o nounset`, but maybe add some explicit validation for them? Just to make debugging easier.

File: hack/e2e-common.sh
Comment (2uasimojo) [inline hack/e2e-common.sh:317]: Similar here for `$SHARED_DIR`, though the `-f` below ought to catch if it's unset (unless we get unlucky and there's a `/vips.txt` in the file system...)

**Review by 2uasimojo — APPROVED** [review]
/lgtm
/hold

I don't mind merging this to get moving; unhold if you want to do that and fix the noted issues later.

**Comment by 2uasimojo** [comment]
/test e2e
/lgtm
/override "Red Hat Konflux / hive-operator-enterprise-contract / hive"

**Comment by lleshchi** [comment]
@2uasimojo ready for review

**Comment by 2uasimojo** [comment]
> @2uasimojo ready for review

You have my lgtm -- unhold at will.

**Comment by lleshchi** [comment]
/hold cancel
/override "Red Hat Konflux / hive-operator-enterprise-contract / hive"

**Pull request merged**
